### PR TITLE
Add Tag Schema And create new tag Route

### PR DESCRIPTION
**Pull request by YJDoc2**
Made basic Tag schema in Models, Added routes file in controllers for tags, added create new Tag route.

### CODE REVIEW

**Comment by vixrant** [comment]
Related to #3.

File: controllers/index.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 const userRouter = require('./user');
+const TagRouter = require('./tag');
```
Comment (vixrant) [inline controllers/index.js]: Use camelCase

File: models/tag.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    type: String,
+    required: [true, 'Please Provide a Display Name for tag'],
+  },
+
+  isSchool: {
+    type: Boolean,
+    default: false,
+  },
+  //* Alternate Versions for the tag, Eg Artificial Intelligence can be alt as AI or ai or A.I.
+  alt: {
```
Comment (vixrant) [inline models/tag.js]: Rename it to "slug". Slug is more commonly used name for such attribute.

File: controllers/tag.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const Tag = require('../models/tag');
+
+const router = express.Router();
+
+router.post('/new', async (req, res) => {
```
Comment (vixrant) [inline controllers/tag.js]: Let it be `/` instead of `/new`. More standard REST API format.

File: models/tag.js (hunk 2 above)
Comment (YJDoc2) [reply models/tag.js]: Change 'name' or 'alt'?

Comment (vixrant) [reply models/tag.js]: Change "alt" -> "slug", even in the default function below.

**Pull request merged**